cloud/scope/vpc_cluster.go

```diff
@@ -221,6 +222,89 @@ func (s *VPCClusterScope) CheckTagExists(tagName string) (bool, error) {
+	// TODO(cjschaef): Add logic to handle cases not default.
+	return infrav1beta2.DefaultAPIServerPort
+}
+
+// GetControlPlaneSubnetIDs returns all of the Control Plane subnet Id's.
+func (s *VPCClusterScope) GetControlPlaneSubnetIDs() ([]string, error) {
+	subnets := make([]string, 0)
+	// Retrieve the subnet Id's from Status.
+	if s.NetworkStatus() != nil && s.NetworkStatus().ControlPlaneSubnets != nil {
+		for _, subnet := range s.NetworkStatus().ControlPlaneSubnets {
+			subnets = append(subnets, subnet.ID)
+		}
+		// NOTE(cjschaef): We assume all Subnets are in Status at this point, we could perhaps reconcile Status with any defined in Spec (preventing duplicates) to be safe.
+		return subnets, nil
+	}
+
+	// NOTE(cjschaef): If Status was not set or ControlPlaneSubnets was empty, the Control Plane subnet ID's could be retrieved from Spec. However, for now consider this an error, since Subnet reconciliation should have run prior and no tracked Control Plane subnets would be a major issue.
+	return subnets, fmt.Errorf("error no control plane subnets available in status")
+}
+
+// GetLoadBalancerHostName will return the hostname of the cluster's public Load Balancer, assuming only one public Load Balancer was provided. Or, the hostname of the single private Load Balancer (assuming the cluster has no public access and only one private Load Balancer was provided).
+// NOTE(cjschaef): A webhook validation check could help ensure this.
+func (s *VPCClusterScope) GetLoadBalancerHostName() (*string, error) {
+	// If no Status or Load Balancer Status is populated, assume the Load Balancer's are not ready (have not been reconciled), so no hostname will be available.
+	if s.NetworkStatus() == nil || s.NetworkStatus().LoadBalancers == nil || len(s.NetworkStatus().LoadBalancers) == 0 {
+		return nil, nil
+	}
+
+	// If no Load Balancer's were defined, return an error, as a Load Balancer must be defined (no default Load Balancer is supported currently).
+	if len(s.NetworkSpec().LoadBalancers) == 0 {
+		return nil, fmt.Errorf("error no load balancers defined for cluster")
+	}
+
+	// This function will check if the Load Balancer has an ID set, use that to check Status for the Load Balancer's hostname. Or, if Name is non-empty, lookup the current Load Balancer status for hostname. Other cases result in errors.
+	getHostName := func(loadBalancer infrav1beta2.VPCLoadBalancerSpec) (*string, error) {
+		// Attempt to retrieve the Load Balancer's hostname from Status by ID.
+		if loadBalancer.ID != nil {
+			if lb, ok := s.NetworkStatus().LoadBalancers[*loadBalancer.ID]; ok {
+				return lb.Hostname, nil
+			}
+			// If the Load Balancer isn't in Status, assume it is not ready/reconciled.
+			return nil, nil
+		}
+
+		// NOTE(cjschaef): As a default Load Balancer isn't supported (when none are provided in Spec), this should result in an error.
+		// TODO(cjschaef): Likely would be beneficial to have webhook validation for this case.
+		if loadBalancer.Name == "" {
```

Comment:
Not sure this check is needed here? in `getLoadBalancer()` we get the default name incase of name is empty.

Reply:
I'll be refactoring, will see what makes sense.